minpy/visualize/event_file_writer.py

```diff
@@ -0,0 +1,230 @@
+"""Writes events to a disk in a logdir."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import threading
+import time
+import socket
+import logging
+import six
+
+import minpy.visualization.event_pb2 as event_pb2
+
+import tensorflow as tf
+
+def as_bytes(bytes_or_text, encoding='utf-8'):
+    """Converts either bytes or unicode to `bytes`, using utf-8 encoding for text.
+    Args:
+        bytes_or_text: A `bytes`, `str`, or `unicode` object.
+        encoding: A string indicating the charset for encoding unicode.
+    Returns:
+        A `bytes` object.
+    Raises:
+        TypeError: If `bytes_or_text` is not a binary or unicode string.
+    """
+    if isinstance(bytes_or_text, six.text_type):
+        return bytes_or_text.encode(encoding)
+    elif isinstance(bytes_or_text, bytes):
+        return bytes_or_text
+    else:
+        raise TypeError('Expected binary or unicode string, got %r' %
+                        (bytes_or_text,))
+
+
+def directorycheck(path):
+    # If the direcotry does not exist, create it!
+    if not os.path.exists(path):
+        os.makedirs(path)
+    # Else, empty the directory.
+    else:
+        fileList = os.listdir(path)
+        for file in fileList:
+            os.remove(path + "/" + file)
+
+
+class EventsWriter:
+    """
+    @@__init__
+
+    @@WriteEvent
+
+    @@_WriteSerializedEvent
+    """
+
+    def __init__(self, file_prefix):
+        '''Events files have a name of the form
+        '/some/file/path/events.out.tfevents.[timestamp].[hostname]'
+        '''
+        self._file_prefix = file_prefix + "out.tfevents." \
+                            + str(time.time())[:10] + "." + socket.gethostname()
+
+        # Open(Create) the log file with the particular form of name.
+        logging.basicConfig(filename=self._file_prefix)
+
+        self._num_outstanding_events = 0
+
+        self._recordio_writer = tf.python_io.TFRecordWriter(self._file_prefix)
+
+        # Initialize an event instance.
+        self._event = event_pb2.Event()
+
+        # In events_writer.cc, event.set_wall_time(time_in_seconds) was
+        # implemented, which I did not mangage to figure out where it was
+        # defined. I assumed this directly modify event.wall_time.
+        self._event.wall_time = time.time()
+
+        # Similar as above, event.set_file_version is assumed to be modifying
+        # the file_version of an event instance. As asserted in event.proto
+        # and events_writer.h, current version is "brain.Event:2" and all
+        # versions start with "brain.Event:".
+        self._event.file_version = "brain.Event:2"
+        self.WriteEvent(self._event)
+
+    def WriteEvent(self, event):
```

Comment:
Change the name of these functions to more pythonic style (no uppercases, underscores between words).